Output broken for multiline messages

Given following test case 

```yaml
- case: break_following
  main: |
    from typing import overload, TypeVar

    reveal_type(1 + 1)  # N: Revealed type is "builtins.int"

    T = TypeVar("T", int, str)
    
    @overload
    def f(x: int) -> int: ...
    @overload
    def f(x: str) -> str: ...
    def f(x: T) -> T:
      return x

    reveal_type(f([1, 2, 3]))  # N: Revealed type is "Any"
```

will get message as follows:

```
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output: 
E   Expected:
E     main:3: note: Revealed type is "builtins.int"
E     <45 (diff)
E   Actual:
E     main:3: note: Revealed type is "builtins.int"
E     main:14: error: No overload variant of "f" matches argument type "List[int]" (diff)
E     main:14: note:     def f(x: int) -> int       (diff)
E     main:14: note:     def f(x: str) -> str       (diff)
E     main:14: note: Possible overload variants:    (diff)
E     main:14: note: Revealed type is "Any"         (diff)
E   
E   Alignment of first line difference:
E     E: main:14: note: Revealed type is "Any"...
E     A: main:14: error: No overload variant of "f" matches argument type "List[i...
E    
```

while mypy will return 

```
main.py:13: error: No overload variant of "f" matches argument type "List[int]"
main.py:13: note: Possible overload variants:
main.py:13: note:     def f(x: int) -> int
main.py:13: note:     def f(x: str) -> str
main.py:13: note: Revealed type is "Any"
```

Please note that

```
E     main:14: note: Possible overload variants:    (diff)
```
is misplaced compared to `mypy` output.

It happens, because sorting key includes actual error message:

https://github.com/typeddjango/pytest-mypy-plugins/blob/f7b249d0e40c25e2979adfbdb804a2e489072837/pytest_mypy_plugins/utils.py#L209

so sort is not stable within (`fname`, `line_number`).


Environment:

- Python: 3.9.7
- pytest-mypy-plugins: 1.9.1
- mypy: 0.910

--- a/pytest_mypy_plugins/tests/test_utils.py
+++ b/pytest_mypy_plugins/tests/test_utils.py
@@ -9,6 +9,7 @@ from pytest_mypy_plugins.utils import (
     TypecheckAssertionError,
     assert_expected_matched_actual,
     extract_output_matchers_from_comments,
+    sorted_by_file_and_line,
 )
 
 
@@ -184,3 +185,35 @@ def test_assert_expected_matched_actual_failures(
         assert_expected_matched_actual(expected, actual_lines)
 
     assert e.value.error_message.strip() == expected_error_message.strip()
+
+
+@pytest.mark.parametrize(
+    "input_lines",
+    [
+        [
+            '''main:12: error: No overload variant of "f" matches argument type "List[int]"''',
+            """main:12: note: Possible overload variants:""",
+            """main:12: note:     def f(x: int) -> int""",
+            """main:12: note:     def f(x: str) -> str""",
+        ],
+        [
+            '''main_a:12: error: No overload variant of "g" matches argument type "List[int]"''',
+            '''main_b:12: error: No overload variant of "f" matches argument type "List[int]"''',
+            """main_b:12: note: Possible overload variants:""",
+            """main_a:12: note:     def g(b: int) -> int""",
+            """main_a:12: note:     def g(a: int) -> int""",
+            """main_b:12: note:     def f(x: int) -> int""",
+            """main_b:12: note:     def f(x: str) -> str""",
+        ],
+    ],
+)
+def test_sorted_by_file_and_line_is_stable(input_lines: List[str]) -> None:
+    def lines_for_file(lines: List[str], fname: str) -> List[str]:
+        prefix = f"{fname}:"
+        return [line for line in lines if line.startswith(prefix)]
+
+    files = sorted({line.split(":", maxsplit=1)[0] for line in input_lines})
+    sorted_lines = sorted_by_file_and_line(input_lines)
+
+    for f in files:
+        assert lines_for_file(sorted_lines, f) == lines_for_file(input_lines, f)

--- a/pytest_mypy_plugins/utils.py
+++ b/pytest_mypy_plugins/utils.py
@@ -194,15 +194,15 @@ def remove_empty_lines(lines: List[str]) -> List[str]:
 
 
 def sorted_by_file_and_line(lines: List[str]) -> List[str]:
-    def extract_parts_as_tuple(line: str) -> Tuple[str, int, str]:
+    def extract_parts_as_tuple(line: str) -> Tuple[str, int]:
         if len(line.split(":", maxsplit=2)) < 3:
-            return "", 0, ""
+            return "", 0
 
-        fname, line_number, contents = line.split(":", maxsplit=2)
+        fname, line_number, _ = line.split(":", maxsplit=2)
         try:
-            return fname, int(line_number), contents
+            return fname, int(line_number)
         except ValueError:
-            return "", 0, ""
+            return "", 0
 
     return sorted(lines, key=extract_parts_as_tuple)
 

